src/ray/gcs/redis_context.cc

```diff
@@ -303,11 +303,15 @@ Status RedisContext::RunAsync(const std::string &command, const UniqueID &id,
 
 Status RedisContext::SubscribeAsync(const ClientID &client_id,
                                     const TablePubsub pubsub_channel,
-                                    const RedisCallback &redisCallback) {
+                                    const RedisCallback &redisCallback,
+                                    int64_t *out_callback_index) {
   RAY_CHECK(pubsub_channel != TablePubsub::NO_PUBLISH)
       << "Client requested subscribe on a table that does not support pubsub";
 
   int64_t callback_index = RedisCallbackManager::instance().add(redisCallback);
+  if (out_callback_index != nullptr) {
```

Comment:
Why would this ever equal `nullptr`? Can we instead make this `RAY_CHECK(out_callback_index != nullptr)`?

Reply:
I changed to `RAY_CHECK(out_callback_index != nullptr)`. However, I think pointer means that the caller can leave it a `nullptr`. If the caller does not need this index, he/she does not need to create a variable to pass in and then abandon it.